libs/payments/ui/src/lib/nestapp/nextjs-actions.service.ts

```diff
@@ -343,4 +345,50 @@ export class NextJSActionsService {
       currency,
     };
   }
+
+  @SanitizeExceptions()
+  @NextIOValidator(DetermineLocationActionArgs, DetermineLocationActionResult)
+  @WithTypeCachableAsyncLocalStorage()
+  async determineLocation(args: {
+    sanctionedLocations: string;
+    offeringId: string;
+    acceptLanguage?: string | null;
+    selectedLanguage?: string;
```

Comment:
nit. The `offering.countries` don't change based on locale, so it'd be fine to provide a default value for these or even remove the args completely.